tests/test_subscriber.py

```diff
@@ -193,19 +195,114 @@ def test_on_put_invalid(self, mock_invalid_param):
         self.assertTrue('invalid_param' in str(context.exception))
 
 
+@patch.object(Persister, 'create_indexes')
+@patch.object(Config, 'load_defaults')
+@patch('history.subscriber.persister.Messenger')
+def test_persist_all_events(mock_messenger, mock_config, mock_create_indexes):
+
+    from history.subscriber.persister import start_dojot_messenger
```

Comment:
Here you can do something like, check if messenger.create_channel was called with the parameters:

config.dojot['subjects']['devices'], "r"
config.dojot['subjects']['device_data'], "r"
"dojot.notifications", "r"

You could also check that the calls in messenger.on are called properly with the expected parameters.

And if messenger.create_channel is called 3 times and messenger.on is called 4 times